airflow/jobs/backfill_job.py

```diff
@@ -946,3 +952,12 @@ def query(result, items):
 
         self.log.info("Reset the following %s TaskInstances:\n\t%s", len(reset_tis), task_instance_str)
         return len(reset_tis)
+
+    def notify_dagrun_state_changed(self, dag_run: DagRun, msg: str = ""):
+        if dag_run.state == DagRunState.RUNNING:
+            get_listener_manager().hook.on_dag_run_running(dag_run=dag_run, msg=msg)
+        elif dag_run.state == DagRunState.SUCCESS:
+            get_listener_manager().hook.on_dag_run_success(dag_run=dag_run, msg=msg)
+        elif dag_run.state == DagRunState.FAILED:
+            get_listener_manager().hook.on_dag_run_failed(dag_run=dag_run, msg=msg)
+        # deliberately not notifying on QUEUED
```

Comment:
This shouldn't be job specific right? Can't this live directly in DagRun itself?

Reply:
Fixed.